tools/secure_notebook.py

```diff
@@ -66,16 +78,33 @@ def create_self_signed_cert(cert_dir, keyfile, certfiile):
     except ConfigFileNotFound:
         config = Config()
 
-    config.NotebookApp.password = pw
+    yield config
 
     with io.open(os.path.join(jupyter_config_dir(), 'jupyter_notebook_config.json'), 'w') as f:
         f.write(six.u(json.dumps(config, indent=2)))
 
-    print('... done')
-    print()
 
-    print("Now let's generate self-signed certificates to secure your connexion.")
+def set_password():
+    """Ask user for password, store it in notebook json configuration file"""
+
+    print("first choose a password.")
+    pw = passwd()
+    print("We will store your password encrypted in the notebook configuration file: ")
+    print(pw)
+
+    with load_config() as config:
+        config.NotebookApp.password = pw
+
+    print('... done\n')
+
+
+def set_certifs():
+    """
+    generate certificate to run notebook over ssl and set up the notebook config.
+    """
+    print("Let's generate self-signed certificates to secure your connexion.")
```

Comment:
Likely, in the end this is in tools, we could migrate it to the notebook itself later, but it does not replace a real audited way of having a SSL certificate and secure connexion.